p2p/raintree/network.go

```diff
@@ -28,13 +28,7 @@ type rainTreeNetwork struct {
 	addrBookProvider addrbook_provider.AddrBookProvider
 
 	peersManager *peersManager
-
-	// TODO (#278): What should we use for de-duping messages within P2P?
-	// TODO(#388): generalize to use the shared `FIFOMempool` type (in `utility/types/mempool.go` at the time of writing) in here as well for this.
-	nonceSet         map[uint64]struct{}
-	nonceList        []uint64
-	mempoolMaxNonces uint64
-	nonceSetMutex    sync.Mutex
+	nonceDeduper *mempool.GenericFIFOSet[uint64, uint64]
```

Comment:
because it's not doing anything "special" just using the basic behaviour offered by the genericFIFOSet. 

In the other cases there are some nuances that I captured creating a specific type.

I am not sure we should add a comment to be honest... Should we really make the reader aware of the fact that there are other implementations of this type that instead are wrapped in a specific type because they have some extra specific behaviour here even if it's completely unrelated with raintree and deduping?